src/core/thread/mle_router.cpp

```diff
@@ -168,6 +173,11 @@ Error MleRouter::SetRouterEligible(bool aEligible)
         break;
 
     case kRoleChild:
+        if (mRouterEligible)
+        {
+            mRouterRoleTransition.StartTimeout();
```

Comment:
I think the router role transition timeout will be started when receiving an MLE Advertisement from parent:

https://github.com/openthread/openthread/blob/43cb7a09cf5b3567246be7f91ba5b9985fe48819/src/core/thread/mle_router.cpp#L1272

Is there a specific issue we're addressing here?

Reply:
Thanks. Good point. I removed the "bug" tag.

This would just help with this behavior in the tests. So we can start the timer earlier (or in more controlled manner) when `SetRouterEligible` is called updating the router eligibility.